huaweicloud/services/dli/resource_huaweicloud_dli_global_variable.go

```diff
@@ -0,0 +1,247 @@
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceGlobalVariable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGlobalVariableCreate,
+		UpdateContext: resourceGlobalVariableUpdate,
+		ReadContext:   resourceGlobalVariableRead,
+		DeleteContext: resourceGlobalVariableDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  `The name of a Global variable.`,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The value of Global variable.`,
+			},
+			"is_sensitive": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Whether to set a variable as a sensitive variable. The default value is **false**.`,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Project ID.`,
+			},
+		},
+	}
+}
+
+func resourceGlobalVariableCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createGlobalVariable: create a Global variable.
+	var (
+		createGlobalVariableHttpUrl = "v1.0/{project_id}/variables"
+		createGlobalVariableProduct = "dli"
+	)
+	createGlobalVariableClient, err := cfg.NewServiceClient(createGlobalVariableProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DLI Client: %s", err)
+	}
+
+	createGlobalVariablePath := createGlobalVariableClient.Endpoint + createGlobalVariableHttpUrl
+	createGlobalVariablePath = strings.ReplaceAll(createGlobalVariablePath, "{project_id}", createGlobalVariableClient.ProjectID)
+
+	createGlobalVariableOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createGlobalVariableOpt.JSONBody = utils.RemoveNil(buildCreateGlobalVariableBodyParams(d))
+	_, err = createGlobalVariableClient.Request("POST", createGlobalVariablePath, &createGlobalVariableOpt)
+	if err != nil {
+		return diag.Errorf("error creating GlobalVariable: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error creating DLI global variable: %s", err)
```
please update it in other places.

Reply:
done, and will fix it in generator